crates/wasmparser/src/validator.rs

```diff
@@ -262,7 +262,7 @@ impl WasmFeatures {
             return Err("reference types support is not enabled");
         }
         match r.heap_type() {
-            HeapType::Concrete(_) => {
+            HeapType::Concrete(_) | HeapType::Exact(_) => {
```

Comment:
Here this'll want to check `self.custom_descriptors()` for the `Exact` use case to gate the usage of `Exact` to only when `custom_descriptors()` is enabled.